dts/arm/microchip/mec172xnsz.dtsi

```diff
@@ -287,71 +287,78 @@
 				status = "disabled";
 			};
 		};
-		gpio_000_036: gpio@40081000 {
-			compatible = "microchip,xec-gpio-v2";
-			reg = < 0x40081000 0x80 0x40081300 0x04
-				0x40081380 0x04 0x400813fc 0x04>;
-			interrupts = <3 2>;
-			gpio-controller;
-			label="GPIO000_036";
-			port-id = <0>;
-			girq-id = <11>;
-			#gpio-cells=<2>;
-		};
-		gpio_040_076: gpio@40081080 {
-			compatible = "microchip,xec-gpio-v2";
-			reg = < 0x40081080 0x80 0x40081304 0x04
-				0x40081384 0x04 0x400813f8 0x4>;
-			interrupts = <2 2>;
-			gpio-controller;
-			label="GPIO040_076";
-			port-id = <1>;
-			girq-id = <10>;
-			#gpio-cells=<2>;
-		};
-		gpio_100_136: gpio@40081100 {
-			compatible = "microchip,xec-gpio-v2";
-			reg = < 0x40081100 0x80 0x40081308 0x04
-				0x40081388 0x04 0x400813f4 0x04>;
-			gpio-controller;
-			interrupts = <1 2>;
-			label="GPIO100_136";
-			port-id = <2>;
-			girq-id = <9>;
-			#gpio-cells=<2>;
-		};
-		gpio_140_176: gpio@40081180 {
-			compatible = "microchip,xec-gpio-v2";
-			reg = < 0x40081180 0x80 0x4008130c 0x04
-				0x4008138c 0x04 0x400813f0 0x04>;
-			gpio-controller;
-			interrupts = <0 2>;
-			label="GPIO140_176";
-			port-id = <3>;
-			girq-id = <8>;
-			#gpio-cells=<2>;
-		};
-		gpio_200_236: gpio@40081200 {
-			compatible = "microchip,xec-gpio-v2";
-			reg = < 0x40081200 0x80 0x40081310 0x04
-				0x40081390 0x04 0x400813ec 0x04>;
-			gpio-controller;
-			interrupts = <4 2>;
-			label="GPIO200_236";
-			port-id = <4>;
-			girq-id = <12>;
-			#gpio-cells=<2>;
-		};
-		gpio_240_276: gpio@40081280 {
-			compatible = "microchip,xec-gpio-v2";
-			reg = < 0x40081280 0x80 0x40081314 0x04
-				0x40081394 0x04 0x400813e8 0x04>;
-			gpio-controller;
-			interrupts = <17 2>;
-			label="GPIO240_276";
-			port-id = <5>;
-			girq-id = <26>;
-			#gpio-cells=<2>;
+		pinctrl: pin-controller@40081000 {
+			compatible = "microchip,xec-pinctrl";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0x40081000 0x1000>;
+
+			gpio_000_036: gpio@40081000 {
```

Comment:
Why are the GPIO controllers being moved into the pin control node? I am surprised by this. The two subsystems are relatively independent within zephyr and I'm not aware of any other devicetrees that behave this way. Please justify this.

Reply:
The gpio_000_xxx nodes are accessed by the pinctrl driver to configure the pin. Earlier we used the node handles directly, but it was suggested to use phandle instead in this [comment](https://github.com/zephyrproject-rtos/zephyr/pull/41888#discussion_r791047583).
Hence we updated to use phandle and moved the gpio_000_xxx nodes.

Reply:
The same happens for STM32/GD32 (inherited from Linux, see https://elixir.bootlin.com/linux/latest/source/arch/arm/boot/dts/stm32mp131.dtsi#L167). In such case, there's no pinctrl device, really. It's all done via the GPIO registers.

Reply:
OK, thanks for the justification. This is resolved.

Reply:
As per latest feedback from @gmarull we don't need the gpio_xxx_xxx nodes inside the pinctrl node. We are going to our original approach where we don't use phandle.
Hence we will be moving the gpio_xxx_xxx nodes outside the pinctrl node.